### DISCUSSION TITLE
Open & Edit a .py File as a Notebook (Similar to Databricks Notebooks)

### DISCUSSION THREAD

**Discussion by RLashofRegas** [microsoft/vscode-jupyter#11281] [New Ideas]
I’m surprised I haven’t seen a discussion or issue on this, but please point me to it if there is one.

I am aware of the following existing “integrations” between python files and Jupyter notebooks in VS Code (please correct me if I’m wrong or missing one):

1. Save/convert py file to notebook
2. Save/convert notebook to py file
3. Add magics to separate “cells” and run code interactively from a .py file.

(3) is nice, but it has the following limitations in my opinion:

1. no editing of commands - once the command is executed it’s read-only, this tends to make a very messy interactive window if you are iteratively testing a given block of code with small edits.
2. Interactive window is open as it’s own tab - Given that VS Code still doesn’t support floating tabs, this is a significant limitation because it takes up valuable screen real estate to have both your script and interactive window open side by side.

If you’ve ever used Databricks notebooks they kind of give you the best of both worlds - the editor shows up as a jupyter-like notebook interface, but all of your edits/saves are actually reflected as a .py file making it much easier for source control and when developing production code. In my opinion, this seamless experience is really the holy grail - an option when you right-click a .py file to “Open as Notebook”. Instead of creating a jupyter notebook (at least not that you see), it simply displays the Python file as a notebook interface and automatically inserts the magics for cell separations and such as you work with the Jupyter interface. You get the editing/experimenting features of Jupyter, but at the end of the day you are left with a simple python file rather than having to convert back and forth.

Has a feature like this been discussed before? Are there any plans to implement this?

**Comment by DonJayamanne** [comment]
please upvote this issue
https://github.com/microsoft/vscode-jupyter/issues/1237

that's where we track this feature request

**Reply by RLashofRegas** [reply]
Thanks, can you help me understand how that issue covers this? That issues seems to be related to markdown-based notebooks?

**Reply by DonJayamanne** [reply]
thats the issue we use to track eidting plain text files as notebooks

**Reply by DonJayamanne** [reply]
Another similar & related issue is https://github.com/microsoft/vscode-jupyter/issues/1240
